app/src/main/java/com/group80/uoftinder/RecommendationInteractor.java

```diff
@@ -0,0 +1,82 @@
+package com.group80.uoftinder;
+
+import com.group80.uoftinder.entities.User;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+public class RecommendationInteractor {
+
+//    private User currentUser;
+    private List<User> compatibilityList;
+//    private List<String> compatibilityList;
+
+    public RecommendationInteractor() {
+
+    }
+
+    /**
+     * Mutates the instance variable compatibilityList based on criteria given by
+     * filters parameter and minimum/maximum age values. Also updates compatibilityList
+     * of currentUser, beyond just mutating the instance variable compatibilityList.
+     *
+     * @param filters   A list of sets of integers that represents the wanted criteria.
+     *                  Each set in the list corresponds to a filtering question.
+     *                  For academic users, filters.size() should be 3 (program of study,
+     *                  year of study, and campus). Each integer represents a selected choice.
+     *                  {0, 3, 5} means the first, fourth, and sixth choices were checked.
+     * @param minAge    The minimum age, inclusive
+     * @param maxAge    The maximum age, inclusive
+     */
+    public void filterCompatibilityList(List<Set<Integer>> filters, int minAge, int maxAge) {
+        for(int j = compatibilityList.size() - 1; j >= 0; j--) {
+            User user = compatibilityList.get(j);
+            if(user.getAge() < minAge || user.getAge() > maxAge) {
+                compatibilityList.remove(j);
```

Comment:
I just wanted to make sure that since we are mutating the compatibilityList that the actual compatibilityList is not modified (but only the compatibilityList that is defined in this class). I don't really know how the behaviour will work since I don't see the initializing (or retireval) of the compatibilityList.

Reply:
Good question! My answer is related to the thread right above this one - I think everything will become much more clear when your PR #22 gets merged into main so you can see my changes in context of your `GenerateCompatibilityList` to understand how I was intending to structure the filtering functionality to work :)

Reply:
Yup for sure! When a user decides to filter their feed, they expect to see recommended users who match the filtering criteria they provided. The way we currently show recommended users to them is via compatibility list where the top element gets passed to the Presenter and then the View. In order to show filtered feed, we must filter compatibility list which requires changing the list. Nicholas' comment did raise a reminder to me that maybe we want to preserve the original list in case users want to reset their filters at any time or change their filters - I have a solution in mind for how to approach this, just need to be able to access the code in Nicholas' PR once it is merged. Hope this clarifies things up slightly :)

Reply:
Yep, thanks for the explanation! I was wondering about how you were planning to preserve the original list in case the filter changes, but sounds like you should have that covered.